### PR TITLE
Changing APIs of buildBlobProps and buildUm in RestUtils

### PR DESCRIPTION
**Pull request by vgkholla**
Also adding the Ambry paper to the README.

Changing the API because the whole `RestRequest` is not needed to construct the blob properties and user metadata, only the arguments are required.

**Primary reviewers : @nsivabalan @cgtz**
**Expected time to review: 10 mins**

### CODE REVIEW

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 1]
```diff
@@ -179,7 +179,6 @@ public void getUserMetadataGoodInputTest()
     userMetadata.put(RestUtils.Headers.USER_META_DATA_HEADER_PREFIX + "key1", "value1");
     userMetadata.put(RestUtils.Headers.USER_META_DATA_HEADER_PREFIX + "key2", "value2");
     setUserMetadataHeaders(headers, userMetadata);
-    verifyBlobPropertiesConstructionSuccess(headers);
```
Comment (cgtz) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: Why was this removed?

**Comment by cgtz** [comment]
LGTM, just have one question

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java (hunk 1 above)
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: Removed because the test isn't for verifying blob properties (it is for user metadata).

**Comment by xiahome** [comment]
Could you add some explanation in the description, for the reason to change the API?

**Comment by vgkholla** [comment]
Done.

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 2]
```diff
@@ -620,7 +617,7 @@ private void verifyUserMetadataConstructionSuccess(JSONObject headers, Map<Strin
   // getBlobPropertiesVariedInputTest() helpers.
 
   /**
-   * Verifies that {@link RestUtils#buildBlobProperties(RestRequest)} fails if given a request with bad headers.
+   * Verifies that {@link RestUtils#buildBlobProperties(Map<String,Object>)} fails if given a request with bad headers.
```
Comment (nsivabalan) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: minor: "fails if given bad args"

**Comment by nsivabalan** [comment]
LGTM except one minor comment

**Comment by cgtz** [comment]
LGTM merging

**Comment by vgkholla** [comment]
@cgtz @nsivabalan merge?
